### PR TITLE
Fix authselect invocations to work with 1.0.2

### PR DESCRIPTION
**Pull request by AdamWill**
Since authselect 1.0.2, invoking an authselect command sequence
like this:

['authselect', 'sssd', '', '--force']

does not work: authselect barfs on the empty string arg and
errors out. We must only pass a features arg if we actually have
some text to go in it.

This broke uninstallation.

I don't know for sure whether the fix in `restore` is needed -
it may be that `cfg[1]` is never an empty sequence - but it
seems sensible to do it just in case.

Signed-off-by: Adam Williamson <awilliam@redhat.com>

### CODE REVIEW

**Comment by AdamWill** [comment]
@edewata also ran into this in https://bugzilla.redhat.com/show_bug.cgi?id=1645708#c10 .

File: ipaplatform/redhat/authconfig.py [hunk 1]
```diff
@@ -187,9 +190,10 @@ def restore(self, path):
         if cfg:
             profile = cfg[0]
 
-            cmd = [
-                paths.AUTHSELECT, "select", profile,
-                " ".join(cfg[1]), "--force"]
+            cmd = [paths.AUTHSELECT, "select", profile]
+            if cfg[1]:
+                cmd.append(" ".join(cfg[1]))
```
Comment (tiran) [inline ipaplatform/redhat/authconfig.py]: I think this is also wrong and should ``cmd.extend(cfg[1])``. authselect doesn't complain about the syntax but it behaves differently:

```
# authselect current --raw
sssd with-fingerprint with-silent-lastlog
# authselect select sssd 'with-fingerprint with-silent-lastlog' --force
Backup stored at /var/lib/authselect/backups/2018-11-28-07-54-05.0uFTrd
Profile "sssd" was selected.
The following nsswitch maps are overwritten by the profile:
- passwd
- group
- netgroup
- automount
- services

Make sure that SSSD service is configured and enabled. See SSSD documentation for more information.

# authselect select sssd with-fingerprint with-silent-lastlog --force
Backup stored at /var/lib/authselect/backups/2018-11-28-07-54-29.iNPMzl
Profile "sssd" was selected.
The following nsswitch maps are overwritten by the profile:
- passwd
- group
- netgroup
- automount
- services

Make sure that SSSD service is configured and enabled. See SSSD documentation for more information.
 
- with-fingerprint is selected, make sure fprintd service is configured and enabled

# 
```

Please notice that the second variant prints ``with-fingerprint is selected, make sure fprintd service is configured and enabled`` while the first does not. The second variant is equivalent to ``cmd.extend(cfg[1])``.

**Comment by tiran** [comment]
@AdamWill since the new version of authselect breaks FreeIPA on Fedora 28 and 29, I want to accelerate this fix. I took the liberty to force push the fix:

* I changed ``if cfg[1]: cmd.append(" ".join(cfg[1]))`` to ``cmd.extend(cfg[1])``
* I slightly edited the commit message and included a link to https://pagure.io/freeipa/issue/7776

**Comment by AdamWill** [comment]
Oh, good catch. I just concentrated on staying as similar as possible to the existing code, I didn't notice it was actually doing the wrong thing there. That's really a separate fix, though - it was wrong before as well.

**Comment by AdamWill** [comment]
I was already preparing updates for F28 and F29, note, but I'll revise them to include your adjusted patch.

**Comment by tiran** [comment]
argh, it's even wrong in both cases! Brace yourself for another fix.

**Comment by AdamWill** [comment]
yes. it needs to be fixed *both* ways.

**Comment by AdamWill** [comment]
I guess that works, sure.

**Comment by tiran** [comment]
@AdamWill is not in the list of trusted developers. I added the re-run label to kick off a PR-CI run.

**Comment by AdamWill** [comment]
As Groucho Marx so nearly said, I would never want to work on a project which would have *me* as a trusted developer...:P

**Comment by abbra** [comment]
LGTM and ACK.

**Comment by tiran** [comment]
master:

* 7aceca2da4f4b68b942c796d6152706f12135502 Fix authselect invocations to work with 1.0.2

**Pull request closed without merging**